### ISSUE TITLE
Feature Request: Block caret style for the override mode.

### ISSUE THREAD

**Issue by SalviaSage** [notepad-plus-plus/notepad-plus-plus#4467]
There is a block caret style for the blinking cursor in the insert mode,
But, it is not there for the override mode.

Please also let us make the override cursor look exactly like the block caret in the insert mode.

Forum post: https://notepad-plus-plus.org/community/topic/15693/feature-request-logic-flaw-block-appearence-for-the-override-caret-cursor

**Comment by SalviaSage** [comment]
I made a ticket in their sourceforge page. Hopefully we can get this useful feature.

**Comment by sasumner** [comment]
Note that even if Scintilla decided to implement this change, Scintilla updates are not pulled into Notepad++ on a regular basis.  In fact I believe the version of Scintilla that Notepad++ 7.5.6 uses is several years old!  :(

**Comment by zufuliu** [comment]
This is [implemented](https://sourceforge.net/p/scintilla/feature-requests/1217/) in Scintilla.

If https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/scintilla/version.txt is correct, Notepad++ still use [3.5.6]((https://www.scintilla.org/ScintillaHistory.html)) which is released on 26 May 2015!

**Comment by SalviaSage** [comment]
It looks like this feature is now in the Scintilla API as of Scintilla v4.1.4

We just need the developers of Notepad++ to pull the most recent version of the Scintilla API
into Notepad++ to be able to use this feature.

**Comment by Ekopalypse** [comment]
@SalviaSage npp 7.7 which uses a Scintilla 4.x version is available now. You might give it a try.

**Comment by zufuliu** [comment]
as far as I know, npp still has ui to use CARETSTYLE_OVERSTRIKE_BLOCK, so this is not implemented.

**Comment by sasumner** [comment]
@zufuliu Yes, Notepad++ itself needs to be changed to support it, or a scripting plugin could be used to do it.  It is my understanding that the support is in the version of Scintilla that N++ 7.7+ uses.

**Comment by Ekopalypse** [comment]
@zufuliu, thanks for the comment, as a fact I know that @SalviaSage uses scripting language plugins.
Maybe I should have mentioned it. Like @sasumner posted, with pythonscript plugin you might set it like this.
~~~
CARETSTYLE_OVERSTRIKE_BLOCK = 16
editor.setCaretStyle(CARETSTYLE_OVERSTRIKE_BLOCK )
~~~

**Comment by SalviaSage** [comment]
I just tested that code, and it works.
Thanks for posting it.

**Comment by SalviaSage** [comment]
The latest updates to the Scintilla editor along with Notepad++ broke the following code:
```
CARETSTYLE_OVERSTRIKE_BLOCK = 16
editor.setCaretStyle(CARETSTYLE_OVERSTRIKE_BLOCK )
```
Does anybody know how to bring this option back?

**Comment by rdipardo** [comment]
@SalviaSage, 
your script misunderstands how the API actually works. `CARETSTYLE_OVERSTRIKE_BLOCK` is a bitmask, not a style; correct usage looks like this:

	editor.setCaretStyle(editor.getCaretStyle() | 16)
	
That's (basically) how Notepad2 does it [^1] (as a built-in function), and it works perfectly.

[^1]: https://github.com/zufuliu/notepad2/blob/96175288f641898f9ac72afe192d37056a94893c/src/Styles.c#L1210-L1214
